src/directives/colony/outpostSK.ts

```diff
@@ -12,21 +15,74 @@ export class DirectiveSKOutpost extends Directive {
 	static directiveName = 'outpostSK';
 	static color = COLOR_PURPLE;
 	static secondaryColor = COLOR_YELLOW;
+	static containerRepairTheshold = 0.5;
+
+	memory: DirectiveSKOutpostMemory;
 
 	constructor(flag: Flag) {
 		super(flag, colony => colony.level >= 7);
 	}
+	
+	isCenterRoom(): boolean {
+		const coords = Cartographer.getRoomCoordinates(this.pos.roomName);
+		if(coords.x % 10 != 0 && coords.x % 5 == 0 && coords.y % 10 != 0 && coords.y % 5== 0){
+			return true;
+		}
+		return false;
+	}
 
 	spawnMoarOverlords() {
-		this.overlords.sourceReaper = new SourceReaperOverlord(this);
+		if(this.memory.isCenterRoom == undefined){
+			this.memory.isCenterRoom = this.isCenterRoom();
+		}
+		//skip sourceReapoers for safe SKrooms
+		if(!this.memory.isCenterRoom){
+			this.overlords.sourceReaper = new SourceReaperOverlord(this);
+		}
+	}
+
+	getContainerConstructionSites(): ConstructionSite | undefined {
+		if (!this.pos.isVisible) {
+			return;
+		}
+		const ContainerConstructionSites = _.filter(this.room!.constructionSites, s => s.structureType == STRUCTURE_CONTAINER);
+
+		if(ContainerConstructionSites.length > 0){
+			return ContainerConstructionSites[0];	
+		}			
+		return;
+	}
+
+	getContainersToRepair(): Structure | undefined {
+		if (!this.pos.isVisible) {
+			return;
+		}
+		const containersTorepair = _.filter(this.room!.structures, s => s.structureType == STRUCTURE_CONTAINER && s.hits < DirectiveSKOutpost.containerRepairTheshold * s.hitsMax);
+
+		if(containersTorepair.length > 0){
+			return containersTorepair[0];	
+		}			
+		return;
 	}
 
 	init(): void {
+		// Add this structure/CS to worker overlord's build/repair list
+		if(Game.time % 150 == 0){
+			const containerNeedRepair = this.getContainersToRepair();
+			if (containerNeedRepair && !this.colony.overlords.work.repairStructures.includes(containerNeedRepair)) {
+				this.colony.overlords.work.repairStructures.push(containerNeedRepair);
```

Comment:
sk miners already build and repair their own containers -- why do you want to offload this to the colony workers?

Reply:
drones that harvest minerals do not repair their containers like the drones harvesting energy.
I have noticed in replays, that mineral containers always decays out without repair.

Reply:
this changes does not have a filter for mineral container only, the worker will repair all containers in SKroom below 50%, which will most probably apply to mineral container being below the threshold